provisioner/zalando_eks.go

```diff
@@ -133,6 +142,21 @@ func (z *ZalandoEKSProvisioner) Decommission(
 	)
 }
 
+// NewZalandoEKSModifier returns a new modifier for EKS cluster provisioning,
+// configured to use the given cluster registry.
+func NewZalandoEKSModifier(
+	clusterRegistry registry.Registry,
+) ProvisionModifier {
+	return &ZalandoEKSModifier{
+		clusterRegistry: clusterRegistry,
+	}
+}
+
+// GetPostOptions returns the configuration only known after deploying the first
+// CloudFormation stack.
+//
+// This includes the API server URL, the Certificate Authority data, and the
+// subnets. Additionally GetPostOptions update the cluster
```

Comment:
> Additionally GetPostOptions update the cluster

This comment looks unfinished.

Can we do the config item update outside of this function somewhere? When reading the `GetPostOptions` function name one doesn't expect it to have any side effects (updating the cluster registry).

Reply:
Can we do the config item update outside of this function somewhere? When reading the `GetPostOptions` function name one doesn't expect it to have any side effects (updating the cluster registry).

Reply:
How about we rename the interface method to something like `UpdateProvisioning`? To describe better what the method actually does.

Reply:
Fine. But then I would use the opportunity to rename it further to a more familiar style. How if we change the following to this:

![image](https://github.com/user-attachments/assets/0bf82bcd-aa70-423c-8137-d6d66a42de24)

* Changing `Modifier` to `PostCreationHook`
* Changing `GetPostOptions` to `Execute` or `Call` (e.g. `PostCreationHook.Call()`) which _can_ have side effects and returns `postOptions` (which we should then rename to `hookResult` or something like that)

Reply:
> Changing Modifier to PostCreationHook

My only issue with this name is that we are calling the method not _after_ creating the cluster, but _during_ cluster creation.

Reply:
Since it's just a rename we can also do this in another PR and allow us some time to come up with the proper naming.

Reply:
I updated the interface to `CreationHook`. In reality a provisioner can choose when the hook applies.